go/enclave/events/subscription_manager.go

```diff
@@ -259,6 +259,15 @@ func (s *SubscriptionManager) updateRelevantLogs(logItem *types.Log, userAddrs [
 	}
 }
 
+// Locks the subscription map and retrieves the subscription with subID, or (nil, false) if so such subscription is found.
```

Comment:
was looking at the other usages.
I see we're doing 	`if len(s.subscriptions) == 0 {`
I'd add a read lock to that as well